src/spyglass/image.cr

```diff
@@ -0,0 +1,486 @@
+module Spyglass
+  class Image
+    alias RGBColor = Tuple(Float64, Float64, Float64)
+
+    enum Channel
+      Red
+      Green
+      Blue
+    end
+
+    getter image : LibSod::SodImg
+    getter data : Bottle::Tensor(Float32)
+    getter path : String
+
+    private def initialize(@image, @data, @path)
+    end
+
+    def self.open(filename : String, channels = 0)
+      img = LibSod.sod_img_load_from_file(filename, channels)
+      shape = [img.h, img.w, img.c]
+      strides = [0] * shape.size
+      sz = 1
+
+      shape.size.times do |i|
```

Comment:
I can make this change if you want since I think I added it, but this tensor creation should have its own method, the code is duplicated.